### PR TITLE
Fix dynamic CSS and script loading for FSE

### LINKED ISSUE #652: Blocks CSS not working with FSE
The CSS doesn't work for non-blocks, as in Page Templates and such so we need to test with those. We also need to test if our normal blocks work there.

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #652.

### Summary

Add templates to our pipeline.

### Screenshots 

----

### Test instructions

1. Insert all the blocks.
2. Customize it.
3. Check the CSS in frontend.

#### Query
```javascript
new QueryQA().select('blocks').run()
```

---- 

### Checklist before the final review

- [ ] Visual elements are not affected by independent changes.
- [ ] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [ ] It loads additional script in frontend only if it is required.
- [ ] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [ ] In case of deprecation, old blocks are safely migrated.
- [ ] It is usable in Widgets and FSE.

### CODE REVIEW

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani the PR is ready. You do a final review. Any issue found that is _not_ related to this problem should be separated to that it does hold back this PR.

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel I can see that Button doesn't work properly in FSE + Font Awesome isn't being loaded, neither in FSE nor in a single post.

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel @HardeepAsrani Issues i found in FSE:

- [ ] Font awesome not loading in frontend https://vertis.d.pr/L5AiTc
- [ ] Maps block doesn't work in editor https://vertis.d.pr/so2fZ5
- [ ] Google fonts don't work neither in editor/frontend https://vertis.d.pr/saBzpS
- [ ] The icons for the Sharing icons don't work in frontend (same as the FA problem) https://vertis.d.pr/96eMIp

If they are not related to this PR, you can move them to another issue and this can be moved to **Ready to merge** 👍

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani do you think that FSE ignores this? https://github.com/Codeinwp/otter-blocks/blob/2bf687f8d9ce2a1e88ff72ad9b98d875d3392eb8/inc/class-registration.php#L683-L692

In Post/Page is working as expected.

**Comment by Soare-Robert-Daniel** [comment]
The main problem is this hook: https://github.com/Codeinwp/otter-blocks/blob/14dbef16365d753afde6be3144e999278cfcd8af/inc/class-registration.php#L299

The `subscribe_fa` function is not called thus the loading is not registered.

File: inc/class-registration.php [hunk 1]
```diff
@@ -84,6 +84,7 @@ public function init() {
 		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_block_editor_assets' ) ); // Don't change the priority or else Blocks CSS will stop working.
 		add_action( 'enqueue_block_assets', array( $this, 'enqueue_block_assets' ) );
 		add_filter( 'render_block', array( $this, 'load_sticky' ), 900, 2 );
+		add_filter( 'render_block', [ $this, 'subscribe_fa' ], 900, 2 );
```
Comment (Soare-Robert-Daniel) [inline inc/class-registration.php:87]: By having this hook here, the `subscribe_fa` function is triggered. But the action `add_action(
			'get_footer',` is not getting the updated value.

A better solution is the integration in the pipeline as Templates.

**Comment by Soare-Robert-Daniel** [comment]
@irinelenache can I get a re-test for Fontawesome Icons and Maps?

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel The issue regarding Maps is fixed 🚀 , the others are still happening

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani, I could not find a workaround to fix the google fonts for Frontend.

https://github.com/Codeinwp/otter-blocks/blob/85e9cb1240d837b086bdd93a8fc60be3ca631f36/inc/css/class-block-frontend.php#L107

FSE pages don't have post meta.

**Comment by HardeepAsrani** [comment]
@irinelenache Can you do the QA one last time with this? We think it might work now.

Here are things you should test. Make sure in FSE Frontend, these things work:

- CSS
- Font Awesome
- Fonts

And make sure it works properly in various FSE templates, Header, Footer, Page, Posts, and all that you can try, Query, or whatever or whatever you can think of.

And in the end, make sure things work just as fine with Neve with all these changes.

And that debug.log file stays clean with all this. Thank you!

**Comment by irinelenache** [comment]
@HardeepAsrani  The issues i mentioned are fixed now 🚀 

- [ ] For the icon list, the content color doesn't work neither in editor/frontend https://vertis.d.pr/lOLaNu
- [ ] Some strange issue, if i try to add any Otter block in a Query loop, the editor is getting slower to the point where the page freezes and i have to close it https://vertis.d.pr/SfYDMU

**Comment by HardeepAsrani** [comment]
@irinelenache does it not happen when used with production version from wporg?

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel @HardeepAsrani Tested again and the mentioned issues are fixed, now everything works fine in FSE for me 🚀

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Can you remove the console.log and merge it?

**Pull request merged**